go/pvl/helpers.go

```diff
@@ -214,3 +223,59 @@ func stringsContains(xs []string, x string) bool {
 	}
 	return false
 }
+
+// Check that a url is valid and has only a domain and is not an ip.
+// No port, path, protocol, user, query, or any other junk is allowed.
+func validateDomain(s string) bool {
+	// Throw a protocol in front because the parser wants one.
+	proto := "http"
+	u, err := url.Parse(proto + "://" + s)
+	if err != nil {
+		return false
+	}
+
+	// The final group must include a non-numeric character.
+	// To disallow the likes of "8.8.8.8."
+	dotsplit := strings.Split(strings.TrimSuffix(u.Host, "."), ".")
+	if len(dotsplit) > 0 {
+		hasalpha := regexp.MustCompile(`^.*\D.*$`)
```

Comment:
I think we can drop both the `^$` anchors, and the `.*` wildcards, and match for a single non-digit character anywhere in the string?